In [1]:
import torch
print(torch.cuda.is_available())

True


In [4]:
import huggingface_hub
huggingface_hub.login()

In [3]:
from unsloth import FastModel


model, tokenizer = FastModel.from_pretrained(
    model_name = "MailMindActionItems4",
    max_seq_length = 2046,
    load_in_4bit= True,
    load_in_8bit= False,
    full_finetuning= False,
)

🦥 Unsloth: Will patch your computer to enable 2x faster free finetuning.
🦥 Unsloth Zoo will now patch everything to make training faster!


c:\repos\MailMind\.venv\Lib\site-packages\unsloth_zoo\gradient_checkpointing.py:330: UserWarning: expandable_segments not supported on this platform (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\pytorch\c10/cuda/CUDAAllocatorConfig.h:28.)
  GPU_BUFFERS = tuple([torch.empty(2*256*2048, dtype = dtype, device = f"cuda:{i}") for i in range(n_gpus)])


==((====))==  Unsloth 2025.3.19: Fast Gemma3 patching. Transformers: 4.50.0.dev0.
   \\   /|    NVIDIA GeForce RTX 2060. Num GPUs = 1. Max memory: 6.0 GB. Platform: Windows.
O^O/ \_/ \    Torch: 2.6.0+cu124. CUDA: 7.5. CUDA Toolkit: 12.4. Triton: 3.2.0
\        /    Bfloat16 = FALSE. FA [Xformers = 0.0.29.post3. FA2 = False]
 "-____-"     Free license: http://github.com/unslothai/unsloth
Unsloth: Fast downloading is enabled - ignore downloading bars which are red colored!
Unsloth: Using float16 precision for gemma3 won't work! Using float32.


Using a slow image processor as `use_fast` is unset and a slow processor was saved with this model. `use_fast=True` will be the default behavior in v4.48, even if the model was saved with a slow processor. This will result in minor differences in outputs. You'll still be able to use a slow processor with `use_fast=False`.


In [ ]:
model = FastModel.get_peft_model(
    model,
    finetune_vision_layers     = False, # Turn off for just text!
    finetune_language_layers   = True,  # Should leave on!
    finetune_attention_modules = True,  # Attention good for GRPO
    finetune_mlp_modules       = True,  # SHould leave on always!

    r = 64,           # Larger = higher accuracy, but might overfit
    lora_alpha = 64,  # Recommended alpha == r at least
    lora_dropout = 0,
    bias = "none",
    random_state = 3407,
)

In [ ]:
from unsloth.chat_templates import get_chat_template
tokenizer = get_chat_template(
    tokenizer,
    chat_template = "gemma-3",
)

In [5]:
from datasets import load_dataset
dataset = load_dataset("DenverDawgs18/ActionItems", split="train")

README.md:   0%|          | 0.00/451 [00:00<?, ?B/s]

test-00000-of-00001.parquet:   0%|          | 0.00/270k [00:00<?, ?B/s]

Generating train split:   0%|          | 0/1250 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/250 [00:00<?, ? examples/s]

In [6]:
from unsloth.chat_templates import standardize_data_formats
dataset = standardize_data_formats(dataset)

In [9]:
print(dataset[100])

{'Unnamed: 0': 100, 'Body': 'Looking forward to our discussion.  ', 'Action_Item': 'No action'}


In [7]:
def apply_chat_template(examples):
    formatted_conversations = []
    
    for idx in range(len(examples["Body"])):
        try:
            # Create a proper instruction from the email
            instruction = f"Provide the action item(s) for this email. Action items are defined as things the sender of the email expects the reciever to complete, such as questions and requests. You should try to keep the language the same (for example if it says Tuesday don't return Tues.). Many emails do not have action items, in this case, return No action. Huge mailing lists should never have action items. You need to keep in mind that emails being forwarded or replies to emails that had action items doesn't necessarily mean that the reciever needs to do what the original email says. You should use context to figure that out. \n\n{examples['Body'][idx]}"
            
            # Create a conversation structure for Gemma's chat template
            conversation = [
                {"role": "user", "content": instruction},
                {"role": "assistant", "content": examples["Action_Item"][idx]}
            ]
            
            # Apply the chat template to the conversation
            formatted_text = tokenizer.apply_chat_template(conversation)
            formatted_conversations.append(formatted_text)
        except Exception as e:
            print(f"Error processing example {idx}: {e}")
            # Add a default value for failed processing
            formatted_conversations.append("")
    
    return {"text": formatted_conversations}
dataset = dataset.map(apply_chat_template, batched = True, num_proc=1)
print(dataset[100]["text"])

Map:   0%|          | 0/1250 [00:00<?, ? examples/s]

<bos><start_of_turn>user
Provide the action item(s) for this email. Action items are defined as things the sender of the email expects the reciever to complete, such as questions and requests. You should try to keep the language the same (for example if it says Tuesday don't return Tues.). Many emails do not have action items, in this case, return No action. Huge mailing lists should never have action items. You need to keep in mind that emails being forwarded or replies to emails that had action items doesn't necessarily mean that the reciever needs to do what the original email says. You should use context to figure that out. 

Looking forward to our discussion.<end_of_turn>
<start_of_turn>model
No action<end_of_turn>



In [8]:
from trl import SFTTrainer, SFTConfig
trainer = SFTTrainer(
    model = model,
    tokenizer = tokenizer,
    train_dataset = dataset,
    eval_dataset = None, # Can set up evaluation!
    args = SFTConfig(
        dataset_text_field = "text",
        per_device_train_batch_size = 2,
        gradient_accumulation_steps = 4, # Use GA to mimic batch size!
        warmup_steps = 5,
        num_train_epochs = 10,
        max_steps = 6,
        learning_rate = 2e-4, # Reduce to 2e-5 for long training runs
        logging_steps = 1,
        optim = "adamw_8bit",
        weight_decay = 0.01,
        lr_scheduler_type = "linear",
        seed = 3407,
        report_to = "none", # Use this for WandB etc
        dataset_num_proc= 1,
    ),
)

Unsloth: Switching to float32 training since model cannot work with float16


Unsloth: Tokenizing ["text"]:   0%|          | 0/1250 [00:00<?, ? examples/s]

In [9]:
from unsloth.chat_templates import train_on_responses_only
trainer = train_on_responses_only(
    trainer,
    instruction_part = "<start_of_turn>user\n",
    response_part = "<start_of_turn>model\n",
)

Map (num_proc=16):   0%|          | 0/1250 [00:00<?, ? examples/s]

In [7]:
print(tokenizer.decode(trainer.train_dataset[100]["input_ids"]))

<bos><bos><start_of_turn>user
Provide the action item(s) for this email. Action items are defined as things the sender of the email expects the reciever to complete, such as questions and requests. You should try to keep the language the same (for example if it says Tuesday don't return Tues.). Many emails do not have action items, in this case, return No action. Huge mailing lists should never have action items. You need to keep in mind that emails being forwarded or replies to emails that had action items doesn't necessarily mean that the reciever needs to do what the original email says. You should use context to figure that out. 

Looking forward to our discussion.<end_of_turn>
<start_of_turn>model
No action<end_of_turn>



In [9]:
print(tokenizer.decode([tokenizer.pad_token_id if x == -100 else x for x in trainer.train_dataset[100]["labels"]]).replace(tokenizer.pad_token, " "))

                                                                                                                                                 No action<end_of_turn>



In [10]:
trainer_stats = trainer.train()

==((====))==  Unsloth - 2x faster free finetuning | Num GPUs used = 1
   \\   /|    Num examples = 1,250 | Num Epochs = 1 | Total steps = 6
O^O/ \_/ \    Batch size per device = 2 | Gradient accumulation steps = 4
\        /    Data Parallel GPUs = 1 | Total batch size (2 x 4 x 1) = 8
 "-____-"     Trainable parameters = 119,209,984/4,000,000,000 (2.98% trained)
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`.


Unsloth: Will smartly offload gradients to save VRAM!


Step,Training Loss
1,0.467100
2,0.926000
3,0.279500
4,0.134200
5,0.275600
6,0.060400


In [15]:
from unsloth.chat_templates import get_chat_template
tokenizer = get_chat_template(
    tokenizer,
    chat_template = "gemma-3",
)
messages = [{
    "role": "user",
    "content": [{
        "type" : "text",
        "text" : "Continue the sequence: 1, 1, 2, 3, 5, 8,",
    }]
}]
text = tokenizer.apply_chat_template(
    messages,
    add_generation_prompt = True, # Must add for generation
)
outputs = model.generate(
    **tokenizer([text], return_tensors = "pt").to("cuda"),
    max_new_tokens = 64,  # Increase for longer outputs!
    # Recommended Gemma-3 settings!
    temperature = 1.0, top_p = 0.95, top_k = 64,
)
tokenizer.batch_decode(outputs)

['<bos><start_of_turn>user\nContinue the sequence: 1, 1, 2, 3, 5, 8,<end_of_turn>\n<start_of_turn>model\n13<end_of_turn>']

In [11]:
model.save_pretrained("MailMindActionItems5")
tokenizer.save_pretrained("MailMindActionItems5")

['MailMindActionItems5\\processor_config.json']

In [3]:

from unsloth import FastModel
model, tokenizer = FastModel.from_pretrained(
        model_name = "MailMindActionItems5", # YOUR MODEL YOU USED FOR TRAINING
        max_seq_length = 2048,
        load_in_4bit = True,
        device_map="cuda:0"
        
    )

messages = [{
    "role": "user",
    "content": [{"type" : "text", "text" : f"Provide the action item(s) for this email using context if needed. If not, return No action. Actionable items include questions, requests for comments, and requests for files. {dataset[100]['email']}",}]
}]
text = tokenizer.apply_chat_template(
    messages,
    add_generation_prompt = True, # Must add for generation
)

from transformers import TextStreamer
_ = model.generate(
    **tokenizer([text], return_tensors = "pt").to("cuda"),
    max_new_tokens = 256, # Increase for longer outputs!
    # Recommended Gemma-3 settings!
    temperature = 1.0, top_p = 0.95, top_k = 64,
    streamer = TextStreamer(tokenizer, skip_prompt = True),
)

==((====))==  Unsloth 2025.3.19: Fast Gemma3 patching. Transformers: 4.50.0.dev0.
   \\   /|    NVIDIA GeForce RTX 2060. Num GPUs = 1. Max memory: 6.0 GB. Platform: Windows.
O^O/ \_/ \    Torch: 2.6.0+cu124. CUDA: 7.5. CUDA Toolkit: 12.4. Triton: 3.2.0
\        /    Bfloat16 = FALSE. FA [Xformers = 0.0.29.post3. FA2 = False]
 "-____-"     Free license: http://github.com/unslothai/unsloth
Unsloth: Fast downloading is enabled - ignore downloading bars which are red colored!
Unsloth: Using float16 precision for gemma3 won't work! Using float32.


KeyError: 'email'

In [2]:
from unsloth import FastLanguageModel
from datasets import load_dataset
import evaluate
import torch
from tqdm import tqdm
from collections import defaultdict
import os 
import tempfile


# === Load your fine-tuned Unsloth model ===
model, tokenizer = FastLanguageModel.from_pretrained(
    model_name="MailMindActionItems5",
    max_seq_length=2048,
    load_in_4bit=True,
    device_map="cuda:0"
)

# === Load the dataset ===
dataset = load_dataset("DenverDawgs18/ActionItems", split="test")

# === Prompt template ===
def format_prompt(body):
    return (
        "<|system|>Provide the action item(s) for this email. Action items are defined as things the sender of the email expects the receiver to complete, such as questions and requests. You should try to keep the language the same (for example if it says Tuesday don't return Tues.). Many emails do not have action items, in this case, return No action. Huge mailing lists should never have action items. You need to keep in mind that emails being forwarded or replies to emails that had action items doesn't necessarily mean that the receiver needs to do what the original email says. You should use context to figure that out.<|user|>\n" + body + "\n<|assistant|>"
    )

# === Generation config ===
gen_kwargs = {
    "max_new_tokens": 140,
    "do_sample": False,
    "temperature": 1,
    "top_p": 0.9,
}

# === Initialize tracking ===
all_results = []
tp_results = []  # store results for TPs only

rouge = evaluate.load("rouge")
dataset = list(dataset)

TP = FP = FN = TN = 0

for i, item in enumerate(tqdm(dataset[240:])):
    prompt = format_prompt(item["Body"])
    inputs = tokenizer(prompt, return_tensors="pt").to("cuda")

    with torch.no_grad():
        outputs = model.generate(**inputs, **gen_kwargs)

    decoded = tokenizer.decode(outputs[0], skip_special_tokens=True)
    prediction = decoded.split("<|assistant|>")[-1].strip()
    reference = item["Action_Item"]
    result = rouge.compute(predictions=[prediction], references=[reference])
    # Determine if action items exist
    is_ref_action = reference.strip().lower() != "no action" and reference.strip().lower() != "no action."
    is_pred_action = prediction.strip().lower() != "no action" and prediction.strip().lower() != "no action."

    if is_ref_action and is_pred_action:
        TP += 1
        tp_results.append(result)
    elif not is_ref_action and is_pred_action:
        FP += 1
    elif is_ref_action and not is_pred_action:
        FN += 1
    else:
        TN += 1

    # Always compute ROUGE for overall stats
    if result["rouge1"] <= 0.61:
        print("Index: ", i)
        print('Body: ', item["Body"])
        print("Reference: ", item["Action_Item"])
        print("Predicted: ", prediction)
        for key, value in result.items():
            print(f"{key}: {value:.4f}")

    all_results.append(result)

    # === Intermediate reporting every 10 ===
    if True:
        avg_scores = defaultdict(float)
        for result in all_results:
            for key, value in result.items():
                avg_scores[key] += value
        for key in avg_scores:
            avg_scores[key] /= len(all_results)

        tp_avg_scores = defaultdict(float)
        if tp_results:
            for result in tp_results:
                for key, value in result.items():
                    tp_avg_scores[key] += value
            for key in tp_avg_scores:
                tp_avg_scores[key] /= len(tp_results)

        # Detection performance
        precision = TP / (TP + FP) if (TP + FP) > 0 else 0
        recall = TP / (TP + FN) if (TP + FN) > 0 else 0
        f1 = (2 * precision * recall) / (precision + recall) if (precision + recall) > 0 else 0

        # === Report ===
        print("\n===== METRICS =====")
        print("Overall Average ROUGE:")
        for key, value in avg_scores.items():
            print(f"{key}: {value:.4f}")

        if tp_results:
            print("\nTrue Positive ROUGE:")
            for key, value in tp_avg_scores.items():
                print(f"{key}: {value:.4f}")

        print("\nDetection performance:")
        print(f"TP: {TP}, FP: {FP}, FN: {FN}, TN: {TN}")
        print(f"Precision: {precision:.4f}")
        print(f"Recall:    {recall:.4f}")
        print(f"F1 Score:  {f1:.4f}")


==((====))==  Unsloth 2025.3.19: Fast Gemma3 patching. Transformers: 4.50.0.dev0.
   \\   /|    NVIDIA GeForce RTX 2060. Num GPUs = 1. Max memory: 6.0 GB. Platform: Windows.
O^O/ \_/ \    Torch: 2.6.0+cu124. CUDA: 7.5. CUDA Toolkit: 12.4. Triton: 3.2.0
\        /    Bfloat16 = FALSE. FA [Xformers = 0.0.29.post3. FA2 = False]
 "-____-"     Free license: http://github.com/unslothai/unsloth
Unsloth: Fast downloading is enabled - ignore downloading bars which are red colored!
Unsloth: Using float16 precision for gemma3 won't work! Using float32.


 10%|█         | 1/10 [00:05<00:51,  5.73s/it]


===== METRICS =====
Overall Average ROUGE:
rouge1: 0.7500
rouge2: 0.6667
rougeL: 0.7500
rougeLsum: 0.7500

True Positive ROUGE:
rouge1: 0.7500
rouge2: 0.6667
rougeL: 0.7500
rougeLsum: 0.7500

Detection performance:
TP: 1, FP: 0, FN: 0, TN: 0
Precision: 1.0000
Recall:    1.0000
F1 Score:  1.0000


 20%|██        | 2/10 [01:30<06:56, 52.03s/it]

Index:  1
Body:  ---------------------- Forwarded by Judy Hernandez/HOU/ECT on 08/03/2000=20
01:13 PM ---------------------------


Mail Delivery Subsystem <Mailer-Daemon@postmaster.enron.com> on 08/03/2000=
=20
12:33:35 PM
To: <Judy.Hernandez@enron.com>
cc: =20
Subject: Returned mail: User unknown


The original message was received at Thu, 3 Aug 2000 12:32:57 -0500 (CDT)
from mailman.enron.com [192.168.189.66]

   ----- The following addresses had permanent fatal errors -----
<Priscilla.giLbert@juno.com>

   ----- Transcript of session follows -----
... while talking to mx.jersey.juno.com.:
>>> RCPT To:<Priscilla.giLbert@juno.com>
<<< 550 Priscilla.giLbert@juno.com is not a valid juno user
550 <Priscilla.giLbert@juno.com>... User unknown

   ----- Original message follows -----

Return-Path: <Judy.Hernandez@enron.com>
Received: from mailman.enron.com (mailman.enron.com [192.168.189.66])
 by postmaster.enron.com (8.8.8/8.8.8/postmaster-1.00) with ESMTP id MAA017=
31;
 Thu, 3 Aug 2000 

 30%|███       | 3/10 [01:47<04:11, 35.98s/it]

Index:  2
Body:  Alok Gaur
233 South Wacker Drive
Chicago, IL 60606
agaur@tjbc.coj

To Mr. Ken Lay,

I'm writing to urge you to donate the millions of dollars you made from selling Enron stock before the company declared bankruptcy to funds, such as Enron Employee Transition Fund and REACH, that benefit the company's employees, who lost their retirement savings, and provide relief to low-income consumers in California, who can't afford to pay their energy bills.  Enron and you made millions out of the pocketbooks of California consumers and from the efforts of your employees.

Indeed, while you netted well over a $100 million, many of Enron's employees were financially devastated when the company declared bankruptcy and their retirement plans were wiped out.  And Enron made an astronomical profit during the California energy crisis last year.  As a result, there are thousands of consumers who are unable to pay their basic energy bills and the largest utility in the state is bankrupt.



 40%|████      | 4/10 [02:02<02:46, 27.81s/it]

Index:  3
Body:  Commencing with an NHK (Japanese public TV) report and quickly spreading to 
media inquiries and wire reports, earlier today a story spread through Japan 
that the Enron affiliate E Power has immediate plans to build at least one 
large (1000 MW) power plant.   In fact,  appears that E Power has been 
pursuing this prospect on a smaller scly, but nothing has progressed to high 
level management or board approval.




We are in the process of providing interviews and responding to inquiries 
from major Japanese media and others.  These inquiries have been in 
anticipation of the March 21 partial liberalization of the Japanese power 
sector.  These discussions also are being utilized to confirm that Enron will 
be opening a merchant office in Japan.  This is in addition to the EnCom/E 
Power affiliate, which is focused on industrial power generation.  Attached 
are the Q&As we are using in response to media inquiries.


 
Reference:  No action
Predicted:  *   Confirm tha

 50%|█████     | 5/10 [02:46<02:48, 33.74s/it]

Index:  4
Body:  Rebecca:  Attached, please find  a write up I put together in advance of your 
meeting next week.  I took a crack at answering ExIm's 5 questions but you 
probably have the Enron responses to these on your own.  I have also provided 
sections pertaining to the primary issues ExIm is currently facing, 
initiatives to improve the competitiveness of the institution, and a history 
of Enron's interaction with the bank.  In addition, you should have rec'd a 
fax (about 27 pgs) of a paper Enron helped put together that deals with 
ExIm's competitiveness and suggestions for improvements.

I should be in the office the rest of today (202-466-9152) and can be reached 
over the weekend at my home number (703)761-3760 if you have any questions or 
need additional information.  

In any event, we'll see you at the meeting on Tuesday.

Regards,
Scott

Reference:  Review the write up in advance of your meeting next week. If you need any additional information, call Scott.
Predicted:

 60%|██████    | 6/10 [03:53<02:59, 44.98s/it]

Index:  5
Body:  Attached is the NetWorks term sheet. I understand that a completly new PPM 
should be coming out today.
---------------------- Forwarded by Scott Sefton/HOU/ECT on 07/25/2000 11:01 
AM ---------------------------


Gareth Bahlmann
07/24/2000 02:14 PM
To: Scott Sefton/HOU/ECT@ECT
cc:  
Subject: Term Sheet


---------------------- Forwarded by Gareth Bahlmann/HOU/ECT on 07/24/2000 
02:02 PM ---------------------------


"Astin, Ronald T." <RAstin@velaw.com> on 07/24/2000 11:23:18 AM
To: "'Bolner, Mike (Chase)'" <michael.bolner@chase.com>, "'MacEachron, Susan 
(Chase)'" <Susan.MacEachron@chase.com>, "Walker, Rick (Chase)" 
<richard.walker@chase.com>, "'nancy.marshall@chase.com'" 
<nancy.marshall@chase.com>, "Brown, Bill W. (Director-Enron)" 
<bill.w.brown@enron.com>, "'Coulter, Jodi'" <Jodi.Coulter@enron.com>, 
"Bahlmann, Gareth (Enron)" <gareth.bahlmann@enron.com>, "Osterberg, Edward" 
<eosterberg@velaw.com>, "Ho, Wendy" <who@velaw.com>, "Douglas, Stephen 
(Enron)" <step

 70%|███████   | 7/10 [04:05<01:42, 34.11s/it]

Index:  6
Body:  Vince,

assuming this is fallout from Stinson's departure

will there be any affect on my group?

-Mike

 -----Original Appointment-----
From: 	Crenshaw, Shirley   On Behalf Of Kaminski, Vince J
Sent:	Friday, June 08, 2001 9:14 AM
To:	Lin, Martin; Maureen Raymond (E-mail); Mike A Roberts (E-mail); Osman Sezgen (E-mail); Pinnamaneni Krishnarao (E-mail); Sandeep Kohli (E-mail); Stinson Gibner (E-mail); Tanya Tamarchenko (E-mail); Vasant Shanbhogue (E-mail); Zimin Lu (E-mail)
Subject:	Vince's meeting on Reorganization
When:	Friday, June 15, 2001 10:00 AM-11:00 AM (GMT-06:00) Central Time (US & Canada).
Where:	EB19C2

Reference:  Let Mike know if there will be any effect on his group from Stinson's departure
Predicted:  Mike,

Please let me know if you have any questions regarding the impact on your group.

-Vince
rouge1: 0.4242
rouge2: 0.0645
rougeL: 0.3636
rougeLsum: 0.4242

===== METRICS =====
Overall Average ROUGE:
rouge1: 0.3483
rouge2: 0.2428
rougeL: 0.3396
rougeLsum

 80%|████████  | 8/10 [04:35<01:05, 32.89s/it]

Index:  7
Body:  FYI: Below is some information on the 2nd Annual Texas Finance Festival. 

---------------------- Forwarded by Cindy Justice/HOU/ECT on 03/13/2000 09:52 
AM ---------------------------


"John D. Martin" <J_Martin@baylor.edu> on 03/13/2000 09:43:51 AM
To: Cindy Justice <Cindy.Justice@enron.com>
cc: Bill Petty <Bill_Petty@baylor.edu> 
Subject: Re: Texas Finance Festival


Cindy,

Yes, it's just around the corner and we have our very best program ever.
Our web site is up at

http://hsb.baylor.edu/html/martinj/festival2.html

check it out.  Actually, I was waiting for the final version of the program
from Sheridan to post it before getting in touch with you this week.  We
are planning on a room capacity group of 45-50 folks and are finalizing the
plans for entertainment for Friday and Saturday night on the riverwalk.
Bill is handling the arrangements this year and I'll pass this message on
to him as well.

Do you think we could get an Enron speaker for Saturday's luncheon

 90%|█████████ | 9/10 [04:54<00:28, 28.70s/it]

Index:  8
Body:  ---------------------- Forwarded by Jane M Tholt/HOU/ECT on 11/27/2000 03:38 
PM ---------------------------


"Betonte, Robert - TP2RSB" <RBetonte@socalgas.com> on 11/22/2000 12:41:44 PM
To: "'prez@cook-inlet.com'" <prez@cook-inlet.com>, "'jane.m.tholt@enron.com'" 
<jane.m.tholt@enron.com>, "'gcvenz@duke-energy.com'" 
<gcvenz@duke-energy.com>, "'jtracy@coral-energy.com'" 
<jtracy@coral-energy.com>, "'cbulf@oneok.com'" <cbulf@oneok.com>
cc: "'suyen@cook-inlet.com'" <suyen@cook-inlet.com>, "'wonmi@cook-inlet.com'" 
<wonmi@cook-inlet.com>, "Lorenz, Lad - TPLPL" <LLorenz@socalgas.com>, "Reid, 
Mary - TP5MAD" <MReid@socalgas.com> 
Subject: El Paso FERC Order regarding Topock Allocation




Dear Acquiring Shippers, 

The attached memo discusses the impact of the Topock allocation on your 
rights as acquiring shippers.? Please read the memo and give me a call (213 
244-3832) if I can be of further assistance.? Have a Great Thanksgiving.

??????? ??????? ??????? ??????? ?????

100%|██████████| 10/10 [05:04<00:00, 30.41s/it]

Index:  9
Body:  Is someone from our group working on this?
----- Forwarded by Tana Jones/HOU/ECT on 08/01/2000 01:14 PM -----

	Tracy Ngo
	08/01/2000 07:48 AM
		 
		 To: Tana Jones/HOU/ECT@ECT
		 cc: 
		 Subject: assignment of deals from Merchant Energy Group of the Americas to 
Morgan Stanley

Tana,

Before Carol St. Clair left for maternity leave, she was working on the 
assignment of some financial and gas deals with Christopher Smith (who 
previously worked in our group).  Since Christopher is no longer here and 
with Carol being out, I'm scrambling to find out who in the financial team is 
working on this.

I've left Mark Taylor a message since Carol's voice mail forwards all 
questions to Mark.

Could you let me know if you have any info?

Thx,
Tracyh
Reference:  Let the sender know if someone from the group is working on the assignment of details from Merchant Energy Group of the Americas to Morgan Stanley
Predicted:  Action items:
*   Let Mark Taylor know if you have any infor

FULL METRICS
TP: 137

FP: 50

FN: 33

TN: 32
Detection Performance:

Precision: 0.7326

Recall: 0.8059

F1 Score: 0.7677

Weighted ROUGE (All 250):

ROUGE-1: 0.3467

ROUGE-2: 0.2603

ROUGE-L: 0.3366

ROUGE-Lsum: 0.3376

Weighted True Positive ROUGE:

ROUGE-1: 0.3769

ROUGE-2: 0.2470

ROUGE-L: 0.3590

ROUGE-Lsum: 0.3629



